services/121-service/test/registrations/update-chosen-fsp.test.ts

```diff
@@ -117,8 +129,92 @@ describe('Update chosen FSP of PA', () => {
 
     // Assert
     expect(response.statusCode).toBe(HttpStatus.OK);
-    expect(response.body.financialServiceProviderName).toBe(
+    expect(response.body.programFinancialServiceProviderConfigurationName).toBe(
       newProgramFinancialServiceProviderConfigurationName,
     );
   });
+
+  it('should succeed updating registration to a newly added FSP config of which the name is not the same as the FSP', async () => {
+    // Arrange
+    const payment = 1;
+    await resetDB(SeedScript.nlrcMultiple);
+    await seedIncludedRegistrations(
+      [registrationPvScoped],
+      programIdPv,
+      accessToken,
+    );
+
+    const newProgramFinancialServiceProviderConfigurationName = 'VisaNumberTwo';
+    const fspConfigBody: CreateProgramFinancialServiceProviderConfigurationDto =
+      {
+        name: newProgramFinancialServiceProviderConfigurationName,
+        financialServiceProviderName:
+          FinancialServiceProviders.intersolveVoucherWhatsapp,
+        label: { en: 'Visa Number Two' },
+        properties: [
+          {
+            name: FinancialServiceProviderConfigurationProperties.password,
+            value: 'password',
+          },
+          {
+            name: FinancialServiceProviderConfigurationProperties.username,
+            value: 'username',
+          },
+        ],
+      };
+    await postProgramFinancialServiceProviderConfiguration({
+      programId: programIdPv,
+      body: fspConfigBody,
+      accessToken,
+    });
+
+    const dataUpdate = {
+      programFinancialServiceProviderConfigurationName:
+        newProgramFinancialServiceProviderConfigurationName,
+    };
+    const reason = 'automated test';
+
+    // Act
+    const response = await updateRegistration(
+      programIdPv,
+      registrationPvScoped.referenceId,
+      dataUpdate,
+      reason,
+      accessToken,
+    );
+    await doPayment(
```

Comment:
if a payment is also part of the 'act', should the test description reflect that?